onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2737,6 +2740,98 @@ ORT_API_STATUS_IMPL(OrtApis::Graph_GetParentNode, _In_ const OrtGraph* graph, _O
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::Graph_GetSubGraph, _In_ const OrtGraph* src_graph,
+                    _In_ const OrtArrayOfConstObjects* ort_nodes_container,
+                    _In_ bool copy_in_memory_initializer,
+                    _Outptr_ OrtGraph** dst_graph) {
+  API_IMPL_BEGIN
+  const GraphViewer& graph_viewer = EpGraph::ToInternal(src_graph)->GetGraphViewer();
+
+  // This API constructs an onnxruntime::Graph from scratch using a given set of nodes,
```

Comment:
Do we have to construct a onnxruntime::Graph from scratch? Could we instead reuse the same Graph and construct a new GraphVIewer with a new filter: https://github.com/microsoft/onnxruntime/blob/7fc6235861d0b01fe6bba3e6b2f233f5797a92cf/onnxruntime/core/graph/graph_viewer.cc#L52

Reply:
I think a problem needs to be asked here is do we want to let this API returns a subgraph that is a standalone OrtGraph instance? 

If we want it to be standalone OrtGraph, then reusing the same Graph to construct a new GraphViewer is not achieving the goal, because we need to make sure the same Graph is around in order to access the nodes, meaning the lifetime of the source OrtGraph instance should be longer than the lifetime of the subgraph OrtGraph instance.

However, if we construct a onnxruntime::Graph from scratch and keep it inside the OrtGraph, we can make the OrtGraph instance standalone.

Reply:
Good point about lifetimes. I didn't expect TRT EP to need to store a standalone `OrtGraph` outside of the call to `GetCapability` or `Compile`.